oxo.py

```diff
@@ -1,32 +1,188 @@
+        """
+
+        return self.board[x][y]
+
+        # raise NotImplementedError("TODO: implement get_square")
 
     def set_square(self, x, y, mark):
-        """ If the specified square is currently empty (0), fill it with mark and return True.
-            If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        """
+        If the specified square is currently empty (0), fill it with mark and return True.
+        If the square is not empty, leave it as-is and return False.
+
+        Args:
+            x (int): x position to place mark.
+            y (int): y position to place mark.
+            mark (int): The player's mark to place.
+
+        Return:
+            bool: True if square is empty and mark is placed. False if square is filled already.
+        """
+
+        if self.board[x][y] == 0:
+            self.board[x][y] = mark
+            self.player = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
-        """ If there are still empty squares on the board, return False.
-            If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        """
+        If there are still empty squares on the board, return False.
+        If there are no empty squares, return True.
+
+        Returns:
+            bool: False is board is not yet full. True if board is full.
+        """
+
+        for x in self.board[0]:
+            for y in self.board[1]:
+                if self.board[x][y] == 0:
+                    return False
+
+        return True
+
+    def collect_positions(self, player):
+        """Collects all the positions owned by a player
+
+        Args:
+            player (int): The player to check positions of.
+
+        Returns:
+            played_spaces (list of int): A list of the spaces owned by the specified player in the format: 0, 1...8 (Right to left, top to bottom).
+        """
+
+        played_spaces = []
+
+        for x in range(0, 3):
+            for y in range(0, 3):
+                if self.board[x][y] == player:
+                    position = [x, y]
+
+                    played_spaces.append(self.convert_to_1d_index(position, 3))
+
+        return played_spaces
+
+    def convert_to_1d_index(self, index, height):
+        """
+        Convert the indexes of a 2d array into a single index for a 1d array. Used to convert the x, y of board space into a single position.
+
+        Args:
+            index (list of int): The x, y indexes stored in a list.
+            height (int): The y dimension of the 2d array.
+
+        Returns:
+            new_index (int): The new index for a 1d array.
+        """
+
+        x = index[0]
+        y = index[1]
+
+        new_index = x + (y * height)
+        return new_index
+
+    def compare_lists(self, master_list, list_to_compare):
```

Comment:
`compare_lists` isn't a very descriptive name (and would make me assume you were comparing for equality) -- something like `is_list_subset` would be better